src/realm/impl/sequential_getter.hpp

```diff
@@ -110,8 +111,8 @@ class SequentialGetter : public SequentialGetterBase {
             return global_end - m_leaf_start;
     }
 
-    size_t m_leaf_start;
-    size_t m_leaf_end;
+    size_t m_leaf_start = 0;
+    size_t m_leaf_end = 0;
```

Comment:
I understand your concern, and I very much support the approach when it comes to local variables in a function. On the other side I think that objects should always have a well-defined state. If you can give the member variable a sensible value, I think you should.